src/dataflow/src/source/mod.rs

```diff
@@ -961,10 +967,17 @@ where
                                 return SourceStatus::Alive;
                             }
                         }
-                        Ok(None) => {
+                        Ok(NextMessage::Pending) => {
                             // There were no new messages
                             break;
                         }
+                        Ok(NextMessage::Finished) => {
+                            if let Consistency::RealTime = consistency_info.source_type {
```

Comment:
might want to add a comment explaining the reasoning behind not doing it  for BYO (to the extent that such reasoning exists ...)

up to you though.

Reply:
I agree that would have been smart, but didn't seem worth going back through CI if all of this code is going to get refactored soon.